### PR TITLE
[DET-2250] docs: add docs how-to and experiment configurations for data-layer 

### PR DESCRIPTION
**Pull request by aaron276h**
Tested by reading

### CODE REVIEW

File: docs/how-to/data-layer.txt [hunk 1]
```diff
@@ -0,0 +1,129 @@
...
+
+The data-layer in Determined uses ``name`` and ``version`` keys as unique identifiers for dataset
+caches. When a dataset cache with matching keys is found it's re-used, allowing users to skip
+pre-processing steps.
+
+
+Configuring Data-Layer Storage
+------------------------------
+
+Data-layer can be configured to cache datasets in ``S3``, ``GCS``, or on the agent's file system.
```
Comment (katport) [inline docs/how-to/data-layer.txt:50]: we just say data-layer here versus "the data layer"

File: docs/how-to/data-layer.txt [hunk 2]
```diff
@@ -0,0 +1,127 @@
+.. _data-layer:
+
+Data-Layer
```
[the remark this answers is not in the document]
Comment (katport) [reply docs/how-to/data-layer.txt:3]: So.... I can see it being hyphenated based on the link @aaron276h sent (which was also my first link in google). It could fall under more than one rule if we consider this an "unusual compound nouns": 

> When writing out new, original, or **unusual compound nouns**, writers should hyphenate whenever doing so avoids confusion.

However, we also don't do "random-access-layer" or "sequential-layer". and the following link suggest we shouldn't use one https://www.grammarly.com/blog/hyphen/:

`Open compounds are typically made up of two nouns that are used together to represent a single idea. “Open” means that there is a space between the two words and no hyphen.`

Basically, everywhere says both are acceptable: https://www.grammar-monster.com/lessons/hyphens_in_compound_nouns.htm: `Use a Hyphen to Eliminate Ambiguity`. It comes down to do we think "data layer" is an unusual noun.

[the remark this answers is not in the document]
Comment (katport) [reply docs/how-to/data-layer.txt:3]: If we want to keep consistency with sequential layer, just do data layer. But I think I like how data-layer _looks_ because it's a determined _feature_
